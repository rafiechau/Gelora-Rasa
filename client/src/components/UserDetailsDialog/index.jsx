import PropTypes from 'prop-types';
import {
  Avatar,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  Skeleton,
  useMediaQuery,
  useTheme,
} from '@mui/material';
import config from '@config/index';
import { useState } from 'react';

import { FormattedMessage } from 'react-intl';
import classes from './style.module.scss';

const UserDetailsDialog = ({ open, onClose, user }) => {
  const [loading, setLoading] = useState(true);

  const theme = useTheme();
  const fullScreen = useMediaQuery(theme.breakpoints.down('sm'));

  return (
    <Dialog open={open} onClose={onClose} maxWidth="sm" fullScreen={fullScreen} data-testid="user-details-dialog">
      <DialogTitle className={classes.dialogTitle} data-testid="dialog-title">
        <FormattedMessage id="app_title_user_details" />
      </DialogTitle>
      <div className={classes.imageContainer}>
        {loading && <Skeleton variant="circle" className={classes.skeleton} />}
        {user?.imagePath ? (
          <>
            {loading && <Skeleton variant="circle" className={classes.skeleton} />}
            <img
              src={`${config.api.server}${user?.imagePath}`}
              alt={user?.firstName}
              loading="lazy"
              className={classes.avatar}
              onLoad={() => setLoading(false)}
              data-testid="user-avatar"
            />
          </>
        ) : (
          <Avatar className={classes.avatar} data-testid="user-avatar">
            {user?.firstName?.split(' ')[0][0]} {user?.firstName?.split(' ') > 1 && user?.firstName?.split(' ')[1][0]}
          </Avatar>
        )}
      </div>

      <DialogContent className={classes.dialogContent}>
        <DialogContentText className={classes.dialogText}>
          <FormattedMessage id="app_column_name" />: {user?.firstName} {user?.lastName}
        </DialogContentText>
        {user?.detailEventOrganizer && (
          <>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_name_ktp" />: {user?.detailEventOrganizer.namaLengkap}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>NIK: {user?.detailEventOrganizer.nik}</DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_gender" />: {user?.detailEventOrganizer.jenisKelamin}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_birth_place" />: {user?.detailEventOrganizer.tempatLahir}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_birth_date" />: {user?.detailEventOrganizer.tanggalLahir}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_blood_type" />: {user?.detailEventOrganizer.golonganDarah}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_address" />: {user?.detailEventOrganizer.alamat}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_province" />: {user?.detailEventOrganizer.provinsi}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_city" />: {user?.detailEventOrganizer.kotaKabupaten}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_subdistrict" />: {user?.detailEventOrganizer.kecamatan}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_village" />: {user?.detailEventOrganizer.kelurahanDesa}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_religion" />: {user?.detailEventOrganizer.agama}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_marital_status" />: {user?.detailEventOrganizer.statusPerkawinan}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_job" />: {user?.detailEventOrganizer.statusKerja}
            </DialogContentText>
            <DialogContentText className={classes.dialogText}>
              <FormattedMessage id="user_details_nationality" />: {user?.detailEventOrganizer.kewarganegaraan}
            </DialogContentText>
          </>
        )}
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose} data-testid="close-button">
          Close
        </Button>
      </DialogActions>
    </Dialog>
  );
};

UserDetailsDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  user: PropTypes.object,
};

export default UserDetailsDialog;
