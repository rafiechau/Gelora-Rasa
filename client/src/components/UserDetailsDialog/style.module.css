.dialog-title {
  text-align: center;
  color: #fff;
}

.dialog-content {
  padding-top: 20px;
}

.image-container {
  display: flex;
  justify-content: center;
  margin-bottom: 1rem;
}
.image-container img, .image-container .avatar {
  border-radius: 50%;
  width: 150px;
  height: 150px;
  border: 2px solid #4CAF50;
  background-color: #fff;
}

.dialogContent {
  padding: 20px;
  border-radius: 5px;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  color: #333;
  line-height: 1.6;
}
.dialogContent .dialogText {
  margin-bottom: 10px;
  font-size: 1rem;
}/*# sourceMappingURL=style.module.css.map */