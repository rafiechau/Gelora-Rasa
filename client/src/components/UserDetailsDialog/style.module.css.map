{"version":3,"sources":["style.module.scss","style.module.css"],"names":[],"mappings":"AAOA;EACI,kBAAA;EACA,WAJI;ACFR;;ADQA;EACI,iBAAA;ACLJ;;ADQA;EACI,aAAA;EACA,uBAAA;EACA,mBAAA;ACLJ;ADMI;EACI,kBAAA;EACA,YAAA;EACA,aAAA;EACA,yBAAA;EACA,sBAnBA;ACeR;;ADQA;EACI,aAAA;EACA,kBA5BY;EA6BZ,wCAAA;EACA,WA/BQ;EAgCR,gBAAA;ACLJ;ADOI;EACI,mBAAA;EACA,eAAA;ACLR","file":"style.module.css"}