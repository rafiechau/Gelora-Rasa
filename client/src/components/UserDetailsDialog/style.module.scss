
$dark-grey: #333;
$border-radius: 5px;
$transition-duration: 0.3s;
$primary-color: #4CAF50;
$white: #fff;

.dialog-title {
    text-align: center;
    color: $white;
}
.dialog-content {
    padding-top: 20px;
}

.image-container {
    display: flex;
    justify-content: center;
    margin-bottom: 1rem; 
    img, .avatar {
        border-radius: 50%;
        width: 150px;
        height: 150px;
        border: 2px solid $primary-color;
        background-color: $white;
      }
}

.dialogContent {
    padding: 20px;
    border-radius: $border-radius;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
    color: $dark-grey;
    line-height: 1.6;

    .dialogText {
        margin-bottom: 10px;
        font-size: 1rem;
    }
}