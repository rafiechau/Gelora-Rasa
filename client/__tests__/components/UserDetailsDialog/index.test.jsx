import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import UserDetailsDialog from '@components/UserDetailsDialog';
import { IntlProvider } from 'react-intl';

// Mock user data
const mockUser = {
  firstName: 'John',
  lastName: 'Doe',
  imagePath: '/path/to/image.jpg',
  detailEventOrganizer: {
    namaLengkap: 'John Doe',
    nik: '1234567890',
    jenisKelamin: 'Male',
    tempatLahir: 'City A',
    tanggalLahir: '2000-01-01',
    golonganDarah: 'O',
    alamat: '123 Main St',
    provinsi: 'Province A',
    kotaKabupaten: 'City B',
    kecamatan: 'District A',
    kelurahanDesa: 'Village A',
    agama: 'Religion A',
    statusPerkawinan: 'Single',
    statusKerja: 'Employed',
    kewarganegaraan: 'Country A',
  },
};

const messages = {
  app_title_user_details: 'User Details',
  user_details_name_ktp: 'KTP Name',
  user_details_nik: 'NIK',
  user_details_gender: 'Gender',
  user_details_birth_place: 'Birth Place',
  user_details_birth_date: 'Birth Date',
  user_details_blood_type: 'Blood Type',
  user_details_address: 'Address',
  user_details_province: 'Province',
  user_details_city: 'City/District',
  user_details_subdistrict: 'Subdistrict',
  user_details_village: 'Village',
  user_details_religion: 'Religion',
  user_details_marital_status: 'Marital Status',
  user_details_job: 'Occupation',
  user_details_nationality: 'Nationality',
  app_column_name: 'Name',
};

const defaultLocale = 'en';

describe('UserDetailsDialog', () => {
  const mockOnClose = jest.fn();

  beforeEach(() => {
    render(
      <IntlProvider locale={defaultLocale} messages={messages.en}>
        <UserDetailsDialog open onClose={mockOnClose} user={mockUser} />
      </IntlProvider>
    );
  });

  test('renders UserDetailsDialog component', () => {
    expect(screen.getByTestId('user-details-dialog')).toBeInTheDocument();
    expect(screen.getByTestId('dialog-title')).toHaveTextContent('app_title_user_details');
  });

  test('calls onClose when close button is clicked', () => {
    fireEvent.click(screen.getByText(/Close/i));
    expect(mockOnClose).toHaveBeenCalled();
  });

  test('should match snapshot', () => {
    const dialog = screen.getByTestId('user-details-dialog');
    expect(dialog).toMatchSnapshot();
  });
});
