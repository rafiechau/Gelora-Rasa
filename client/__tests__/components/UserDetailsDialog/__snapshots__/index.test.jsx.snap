// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailsDialog should match snapshot 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="user-details-dialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":r2:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root dialogTitle css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        data-testid="dialog-title"
        id=":r2:"
      >
        app_title_user_details
      </h2>
      <div
        class="imageContainer"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse skeleton css-58ftfq-MuiSkeleton-root"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-pulse skeleton css-58ftfq-MuiSkeleton-root"
        />
        <img
          alt="John"
          class="avatar"
          data-testid="user-avatar"
          loading="lazy"
          src="http://localhost:5000//path/to/image.jpg"
        />
      </div>
      <div
        class="MuiDialogContent-root dialogContent css-ypiqx9-MuiDialogContent-root"
      >
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          app_column_name
          : 
          John
           
          Doe
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_name_ktp
          : 
          John Doe
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          NIK: 
          1234567890
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_gender
          : 
          Male
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_birth_place
          : 
          City A
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_birth_date
          : 
          2000-01-01
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_blood_type
          : 
          O
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_address
          : 
          123 Main St
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_province
          : 
          Province A
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_city
          : 
          City B
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_subdistrict
          : 
          District A
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_village
          : 
          Village A
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_religion
          : 
          Religion A
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_marital_status
          : 
          Single
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_job
          : 
          Employed
        </p>
        <p
          class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 dialogText css-qfso29-MuiTypography-root-MuiDialogContentText-root"
        >
          user_details_nationality
          : 
          Country A
        </p>
      </div>
      <div
        class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          data-testid="close-button"
          tabindex="0"
          type="button"
        >
          Close
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;
